app/jobs/assignment_repo/porter_status_job.rb

```diff
@@ -2,50 +2,52 @@
 
 class AssignmentRepo
   class PorterStatusJob < ApplicationJob
-    class ImportInProgress < StandardError; end
     queue_as :porter_status
-    retry_on Octopoller::TimeoutError, queue: :porter_status
 
     # rubocop:disable MethodLength
     # rubocop:disable AbcSize
     # rubocop:disable CyclomaticComplexity
     def perform(assignment_repo, user)
       github_repository = assignment_repo.github_repository
 
-      error_handler = proc { |error| logger.warn error.to_s }
-
-      result = Octopoller.poll(error_handler: error_handler) do
-        begin
-          progress = github_repository.import_progress[:status]
-          case progress
-          when GitHubRepository::IMPORT_COMPLETE
-            Creator::REPOSITORY_CREATION_COMPLETE
-          when *GitHubRepository::IMPORT_ERRORS
+      begin
+        result = Octopoller.poll(timeout: 30) do
```

Comment:
I like this tidier polling DSL a lot more 🙌 

The only thing I'd suggest to make this snippet even more concise is to use `timeout: 30.seconds` instead of `timeout: 30`. It's pretty obvious to us that this can't be minutes or ms, but being explicit is usually nice. Looking at the octopoller logic, I think passing `30.seconds` will work just fine.